app/src/main/kotlin/crazyboyfeng/accSettings/acc/Command.kt

```diff
@@ -82,16 +87,22 @@ object Command {
         return properties
     }
 
-    suspend fun getVersion(): Pair<Int, String?> {
-        val version = execAcc("version")
-        if (version.startsWith('v')) {
-            try {
-                val versionCode = version.substringAfter('(').substringBefore(')').toInt()
-                val versionName = version.substringAfter('v').substringBefore(' ')
-                return Pair(versionCode, versionName)
-            } catch (e: NumberFormatException) {
-                e.printStackTrace()//may be cause by upgraded acc
+    // TODO Handle exceptions in better way. if dev/.vr25/ delete, app fails to handle it.
+    suspend fun getVersion(context: Context): Pair<Int, String?> {
+        if (isAccInstalled()) {
```

Comment:
`Command.kt` is just a tool class used to wrap commands as methods, it is not recommended to handle business logic here, otherwise circular calls may occur.

Reply:
That's why i put that comment there!

Reply:
All exceptions concerning the operation of instructions are handled at the specific runtime. For example: https://github.com/CrazyBoyFeng/AccSettings/blob/9fb43805944d7a81d1ff28e16c02049e28999d39/app/src/main/kotlin/crazyboyfeng/accSettings/WorkerReceiver.kt#L47